psychopy/sound/backend_ptb.py

```diff
@@ -165,28 +166,28 @@ def _getSimilar(self, sampleRate, channels=-1, blockSize=-1):
         if (sampleRate not in [None, -1, 0] and
                 channels not in [None, -1] and
                 blockSize not in [None, -1]):
-            return self._getStream(sampleRate, channels, blockSize)
+            return self._getStream(sampleRate, channels, blockSize, device_id)
 
-    def _getStream(self, sampleRate, channels, blockSize):
+    def _getStream(self, sampleRate, channels, blockSize, device_id):
         """Strict check for this format or create new
         """
-        label = getStreamLabel(sampleRate, channels, blockSize)
+        label = getStreamLabel(sampleRate, channels, blockSize, device_id)
         # try to retrieve existing stream of that name
         if label in self:
             pass
         # todo: check if this is still needed on win32
         # on some systems more than one stream isn't supported so check
-        elif sys.platform == 'win32' and len(self):
```

Comment:
@janna-stb is this really not needed any longer, or did you just remove it while making changes to `getStreamLabel` ?